### ISSUE TITLE
Computation of S(q) for a list of atomic positions

### ISSUE THREAD

**Issue by rayosborn** [YellProgram/Yell#6]
We are interested in the possibility of using some of the Yell code to calculate diffuse scattering when the explicit atomic positions of a disordered lattice, in our case derived from an ab initio model, are already known for a large cluster. Is that possible using the existing code?

**Comment by aglie** [comment]
Absolutely. So the trick is to define average unit cell as a variant between a molecule and a Void
```
UnitCell
[
  Var = Variant[
    (p=1/2)
    [Fe 1  0   0 0  0.002
     Fe 1  0.2 0 0 0.002]

    (p=1/2)
    Void
  ]
]
```

and add one `SubstitubtionalCorrelation`

```
Correlations
[
  [(0,0,0)
   SubstitutionalCorrelation(Var,Var,1/2)
  ]
]
```

This is equivalent to simply calculating the form-factor of the molecule (cluster). 

If your cluster is big, the calculation might take significant amount of time (the calculation time is proportional to number of atoms squared), in such case I can show you how to use one yet undocumented feature which will make calculations faster.

**Comment by rayosborn** [comment]
Thanks for your quick responses. We've just started testing it on our trial samples, which are quite large, and it does look as if we would benefit from this extra feature you mentioned. Is it easy to describe how to implement it? We have built the application from source so we can modify the code if that is required.

**Comment by aglie** [comment]
> We have built the application from source so we can modify the code if that is required.

Oh, that is impressive. The feature is actually implemented, but it requires some care to use. This is how to rewrite the above example using a slightly faster calculation:

    CalculationMethod exact
    MolecularScatterers [
      Iron_molecule  =  [Fe 1  0   0 0  0.002
                         Fe 1  0.2 0 0 0.002]
    ]
    UnitCell
    [
       Var = Variant[
         (p=1/2)
         Iron_molecule 1  0 0 0  0

         (p=1/2)
         Void
       ]
    ]
     Modes []
    Correlations
    [
      [(0,0,0)
       SubstitutionalCorrelation(Var,Var,1/2)
      ]
    ]

The line `Iron_molecule  = ...` in effect defines `Iron_molecule` as a new atomic type and precalculates its form factor before calculating PDF. In effect, the script above will create 1 pair and not 4 pairs as in the first script.

*Important*:
In order to make sure that the calculation with direct scatterers works correctly the following points should be met:
* the calculation method should be set to `exact`
* the atoms which depend on molecular scatterers can not be moved with `Symmetry` keyword (in effect the scatterer position would be moved, but the form factor will not be translated/rotated)
*  variables can be used for defining the atoms in the molecular scatterer, but if they cannot be refined. This is because the form factor is calculated only once in the beginning of the calculation and is not recalculated in the course of refinement.

Probably at some time you would like to refine the positions of the atoms in the molecular scatterer. In such case just write me, I will modify the program to make it possible to use the refinable variables in the molecular scatterer.

**Comment by aglie** [comment]
A brief question: which program/script did you use for making the reciprocal space reconstruction?

**Comment by jmjwozniak** [comment]
Hi- do you mind if we make a fork of Yell for our experiments?

**Comment by aglie** [comment]
Sure, you can fork it. Do I have to do something?

2015-04-13 21:04 GMT+02:00 Justin M Wozniak <notifications@github.com>:

**Comment by jmjwozniak** [comment]
No, I just thought I'd send a friendly note.

**Comment by aglie** [comment]
Ah, thank you. Good luck with the code. If you need any help, just ask me,
I will help right away.

2015-04-15 17:59 GMT+02:00 Justin M Wozniak <notifications@github.com>:
